### PR TITLE
Add GitHub Actions Python CI

### PR DESCRIPTION
**Pull request by rlad78**
Adds a GitHub Actions script that attempts the following tasks:
 - Checks for dependency resolution on Python 3.7 – 3.10
 - Tests building on the current and previous version of Poetry
 - Linting using flake8 with standard but thorough presets

flake8 is added as an optional dependency that can be imported by using `poetry install -E github`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,11 +25,15 @@ classifiers = [
 python = "^3.7"
 zeep = "^4.1.0"
 requests = "^2.27.1"
+flake8 = {version = "^4.0.1", optional = true}
```
Comment (bradh11) [inline pyproject.toml]: Can you move this to dev-dependencies rather than primary library dependency?

**Review by bradh11 — APPROVED** [review]

**Comment by bradh11** [comment]
I think you'll want to do a few things to resolve the CI failures.  

1. revert the .gitignore, and keep the poetry.lock file
2. check-in the poetry.lock file

Here is a good reference to why it makes sense to include it
https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

That being said, libraries do not 'require' the lock file, so alternatively you could delete the poetry lock file and commit that change.

That should trigger the tests to run again.

**Comment by bradh11** [comment]
ok, made some updates to the pipeline.  It's failing CI, but i know why.  Will add a few tests to get some boilerplate in place.
  I will merge this and submit a new PR for automated tests (via pytest)

File: pyproject.toml (hunk 1 above)
Comment (rlad78) [reply pyproject.toml]: Sorry, should have mentioned that, for Poetry extras to work, the optional dependency cannot be a dev-only dependency. See the following: https://stackoverflow.com/a/66030875/7666672

In hindsight, it's a lot simpler to keep it as a regular dev dependency.

Comment (bradh11) [reply pyproject.toml]: All good.  We have working CI for automated tests now.  Thanks for the PR!   With our merged updates first CI has finished successfully.  🍾

https://github.com/PresidioCode/ciscoaxl/actions/runs/2341883657

**Pull request merged**